src/test/common/paymentPlan.ts

```diff
@@ -0,0 +1,97 @@
+/* tslint:disable:no-unused-expression */
+import { createPaymentPlan } from 'common/paymentPlan'
+import { expect } from 'chai'
+import * as moment from 'moment'
+
+const TOTAL_AMOUNT = 1000
+
+describe('PaymentPlan',() => {
+  describe('getLastPaymentDate', () => {
+    it('should return the last payment date from given fromDate', () => {
+      const instalmentAmount = 10
+      const fromDate = moment('2018-01-01')
+      const numberOfInstalmentsInWeeks = 200
+      const expectedLastPaymentDate = fromDate.clone().add(numberOfInstalmentsInWeeks, 'weeks')
+
+      const frequencyInWeeks = 2
+      const paymentPlan = createPaymentPlan(TOTAL_AMOUNT, instalmentAmount, frequencyInWeeks)
+      const lastPaymentDate = paymentPlan.getLastPaymentDate(fromDate)
+
+      expect(lastPaymentDate.isSame(expectedLastPaymentDate)).to.be.true
+    })
+  })
+
+  describe('getPaymentLength', () => {
+    it('should return a payment length that has no year showing', () => {
+      const instalmentAmount = 105
+      const frequencyInWeeks = 4
+      const paymentPlan = createPaymentPlan(TOTAL_AMOUNT, instalmentAmount, frequencyInWeeks)
+      const paymentLength = paymentPlan.getPaymentLength()
+      expect(paymentLength).to.equal('8 months 3 weeks')
+    })
+
+    it('should return a payment length that has exactly 1 year showing', () => {
```

Comment:
After reading `should return a payment length that has exactly 1 year showing` I thought that test will evaluate scenario where result is equal `1 year`